### PR TITLE
Random test generation constraints on commands (fixes #198)

### LINKED ISSUE #198: Random test generation constraints on commands
Some commands are not allowed to be called after another command.

### PR DESCRIPTION
**Pull request by afsafzal**

### CODE REVIEW

**Comment by afsafzal** [comment]
@ChrisTimperley this is a somehow messy implementation of the constraints on commands for test generation. The format allows you to specify (in the `yaml` file): 1. a list of commands that can come after this command 2. * for saying all commands can come after this, or 3. * followed by a list of commands which stands for everything but <those commands>
We can make it prettier when we have time. I'll create an issue.

**Comment by afsafzal** [comment]
@ChrisTimperley also we need to figure out which commands can be followed by the other ones to complete the `yaml` file!

File: houston/ardu/copter/commands.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ commands:
   - name: MAV_CMD_NAV_WAYPOINT
     id: 16
     generator: circle_based_generator
+    next-allowed:
+      - "*"
+      - MAV_CMD_NAV_TAKEOFF
```
Comment (ChrisTimperley) [inline houston/ardu/copter/commands.yml]: Isn't this line redundant, or have I misunderstood the semantics of `*`?

Comment (afsafzal) [reply houston/ardu/copter/commands.yml]: This basically means "all commands are allowed except `MAV_CMD_NAV_TAKEOFF`"
if * wasn't there it meant "nothing but `MAV_CMD_NAV_TAKEOFF`"

**Pull request merged**